apps/scorecard/server/dataFunctions/processScorecard.js

```diff
@@ -173,95 +177,125 @@ processScorecard = function (plotParams, plotFunction) {
+            threshold = threshold.replace(/\./g, "__DOT__");
             if (
-              scorecardDocument.results.rows[curve.label].data[region][stat][
-                variable
+              scorecardDocument.results.rows[curve.label].data[region][stat][variable][
+                threshold
               ] === undefined
             ) {
-              scorecardDocument.results.rows[curve.label].data[region][stat][variable] =
-                {};
+              scorecardDocument.results.rows[curve.label].data[region][stat][variable][
+                threshold
+              ] = {};
             }
-            fcstLengths.forEach(function (fcstlen, index) {
-              // this is where we will calculate the significances.
-              // get a random number between 0 and 100
-              let sval = 0;
-              const val = Math.floor(Math.random() * 100);
-              // use the random number to generate a weighted number between -2 and 2
-              if (val >= 0 && val < 10) {
-                sval = -2;
-              } else if (val >= 10 && val < 30) {
-                sval = -1;
-              } else if (val >= 30 && val < 70) {
-                sval = 0;
-              } else if (val >= 70 && val < 90) {
-                sval = 1;
-              } else if (val >= 90 && val <= 100) {
-                sval = 2;
-              }
-
+            curve.level.forEach(function (level) {
+              level = level.replace(/\./g, "__DOT__");
               if (
-                scorecardDocument.results.rows[curve.label].fcstlens.includes(fcstlen)
-              ) {
                 scorecardDocument.results.rows[curve.label].data[region][stat][
                   variable
-                ][fcstlen] = sval;
-              } else {
-                // mark this undefined
+                ][threshold][level] === undefined
+              ) {
                 scorecardDocument.results.rows[curve.label].data[region][stat][
                   variable
-                ][fcstlen] = "undefined";
+                ][threshold][level] = {};
               }
+              fcstLengths.forEach(function (fcstlen, index) {
+                fcstlen = fcstlen.replace(/\./g, "__DOT__");
+                // this is where we will calculate the significances.
+                // get a random number between 0 and 100
+                let sval = 0;
+                const val = Math.floor(Math.random() * 100);
+                // use the random number to generate a weighted number between -2 and 2
+                if (val >= 0 && val < 10) {
+                  sval = -2;
+                } else if (val >= 10 && val < 30) {
+                  sval = -1;
+                } else if (val >= 30 && val < 70) {
+                  sval = 0;
+                } else if (val >= 70 && val < 90) {
+                  sval = 1;
+                } else if (val >= 90 && val <= 100) {
+                  sval = 2;
+                }
```

Comment:
This is pretty neat. It could also be factored out into a function to improve the clarity of this code. Though it looks like this is intended to be removed once we get the backend processor going. If so, I wouldn't worry too much about it. 

E.g.

```js
/** return a weighted number between -2 and 2
 *
 * 10% chance of -2 or 2
 * 20% chance of -1 or 1
 * 40% chance of 0
 */
function getWeightedNumber() {
  let weightedVal = 0;
  const randVal = Math.floor(Math.random() * 100);
  // use the random number to generate a weighted number between -2 and 2
  if (randVal >= 0 && randVal < 10) {
    weightedVal = -2;
  } else if (randVal >= 10 && randVal < 30) {
    weightedVal = -1;
  } else if (randVal >= 30 && randVal < 70) {
    weightedVal = 0;
  } else if (randVal >= 70 && randVal < 90) {
    weightedVal = 1;
  } else if (randVal >= 90 && randVal <= 100) {
    weightedVal = 2;
  }

  return weightedVal;
}

// usage
const randomScorecardVal = getWeightedNumber();
}
```

Reply:
Yeah, this is a placeholder that Randy wrote to give us dummy data and will soon be replaced.